### PR TITLE
Remove the GTS BUFR2IODA part of the snow obs prep job

### PR DESCRIPTION
**Pull request by jiaruidong2017**
# Description
This PR removed the GTS BUFR2IODA part of the snow obs prep job, and replaced it with a direct read from BUFR snow data at runtime by the JEDI executable. 

This PR depends on https://github.com/NOAA-EMC/GDASApp/pull/1276

# Type of change

- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? NO 

# How has this been tested?
- Cycled test on Hera

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Comment by aerorahul** [comment]
@jiaruidong2017 
I left a comment in the GDASapp PR that this PR depends on.  It might have have an impact on this PR.  Let me know if what I commented is confusing.

**Comment by jiaruidong2017** [comment]
> @jiaruidong2017 I left a comment in the GDASapp PR that this PR depends on. It might have have an impact on this PR. Let me know if what I commented is confusing.

Thanks @aerorahul for your suggestions.

**Comment by CoryMartin-NOAA** [comment]
This should partially solve https://github.com/NOAA-EMC/global-workflow/issues/2902

File: ush/python/pygfs/task/snow_analysis.py [hunk 1]
```diff
@@ -272,6 +195,19 @@ def initialize(self) -> None:
         save_as_yaml(self.task_config.jedi_config, self.task_config.jedi_yaml)
         logger.info(f"Wrote letkfoi YAML to: {self.task_config.jedi_yaml}")
 
+        # Read and render the GTS_LIST yaml
+        logger.info(f"Reading {self.task_config.GTS_LIST}")
+        gts_config = parse_j2yaml(self.task_config.GTS_LIST, localconf)
+        logger.debug(f"{self.task_config.GTS_LIST}:\n{pformat(gts_config)}")
+
+        # Generate bufr2ioda mapping YAML files
```
Comment (CoryMartin-NOAA) [inline ush/python/pygfs/task/snow_analysis.py]: why do we need to generate this? is it not static?

Comment (jiaruidong2017) [reply ush/python/pygfs/task/snow_analysis.py]: Yes, you are right. I made the changes to both PRs. Thanks @CoryMartin-NOAA

File: parm/config/gfs/config.snowanl [hunk 2]
```diff
@@ -9,6 +9,7 @@ echo "BEGIN: config.snowanl"
 source "${EXPDIR}/config.resources" snowanl
 
 export OBS_LIST="${PARMgfs}/gdas/snow/obs/lists/gdas_snow.yaml.j2"
+export GTS_LIST="${PARMgfs}/gdas/snow/obs/config/bufr2ioda_mapping.yaml.j2"
```
Comment (CoryMartin-NOAA) [inline parm/config/gfs/config.snowanl]: I think I would be more descriptive here and name this `GTS_SNOW_STAGE_YAML` unless other reviewers find this too verbose

Comment (jiaruidong2017) [reply parm/config/gfs/config.snowanl]: Made changes as suggested.

**Review by CoryMartin-NOAA — DISMISSED** [review]

**Comment by CoryMartin-NOAA** [comment]
Will merge GDASApp PR whenever others are ready, then that hash can be updated here before CI testing

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Review by WalterKolczynski-NOAA — DISMISSED** [review]
Approved pending update to GDAS hash.

**Comment by jiaruidong2017** [comment]
@WalterKolczynski-NOAA GDAS hash has been updated. Thanks.

**Comment by WalterKolczynski-NOAA** [comment]
Looks like it failed due to a hardware issue on Hera. Trying again.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**